<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>TextEditor::Internal::BehaviorSettingsWidget</class>
 <widget class="QWidget" name="TextEditor::Internal::BehaviorSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>801</width>
    <height>459</height>
   </rect>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="TextEditor::SimpleCodeStylePreferencesWidget" name="tabPreferencesWidget" native="true"/>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBoxTyping">
       <property name="title">
        <string>Typing</string>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="0" colspan="2">
         <widget class="QCheckBox" name="autoIndent">
          <property name="text">
           <string>Enable automatic &amp;indentation</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QLabel" name="smartBackspaceLabel">
          <property name="text">
           <string>Backspace indentation:</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="smartBackspaceBehavior">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;
Specifies how backspace interacts with indentation.

&lt;ul&gt;
&lt;li&gt;None: No interaction at all. Regular plain backspace behavior.
&lt;/li&gt;

&lt;li&gt;Follows Previous Indents: In leading white space it will take the cursor back to the nearest indentation level used in previous lines.
&lt;/li&gt;

&lt;li&gt;Unindents: If the character behind the cursor is a space it behaves as a backtab.
&lt;/li&gt;
&lt;/ul&gt;&lt;/body&gt;&lt;/html&gt;
</string>
          </property>
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Follows Previous Indents</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Unindents</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QLabel" name="tabKeyBehaviorLabel">
          <property name="text">
           <string>Tab key performs auto-indent:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0">
         <spacer name="horizontalSpacer_5">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>30</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="tabKeyBehavior">
          <property name="sizePolicy">
           <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <item>
           <property name="text">
            <string>Never</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Always</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>In Leading White Space</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>28</width>
         <height>13</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QGroupBox" name="groupBoxStorageSettings">
       <property name="toolTip">
        <string>Cleanup actions which are automatically performed right before the file is saved to disk.</string>
       </property>
       <property name="title">
        <string>Cleanups Upon Saving</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="0" colspan="2">
         <widget class="QCheckBox" name="cleanWhitespace">
          <property name="toolTip">
           <string>Removes trailing whitespace upon saving.</string>
          </property>
          <property name="text">
           <string>&amp;Clean whitespace</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>30</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="inEntireDocument">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="toolTip">
           <string>Cleans whitespace in entire document instead of only for changed parts.</string>
          </property>
          <property name="text">
           <string>In entire &amp;document</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QCheckBox" name="cleanIndentation">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="toolTip">
           <string>Corrects leading whitespace according to tab settings.</string>
          </property>
          <property name="text">
           <string>Clean indentation</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QCheckBox" name="addFinalNewLine">
          <property name="toolTip">
           <string>Always writes a newline character at the end of the file.</string>
          </property>
          <property name="text">
           <string>&amp;Ensure newline at end of file</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBoxEncodings">
       <property name="title">
        <string>File Encodings</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="encodingLabel">
          <property name="text">
           <string>Default encoding: </string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="encodingBox">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="sizeAdjustPolicy">
           <enum>QComboBox::AdjustToMinimumContentsLengthWithIcon</enum>
          </property>
          <property name="minimumContentsLength">
           <number>20</number>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <spacer name="horizontalSpacer_6">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>285</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="utf8BomBox">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;
&lt;p&gt;How text editors should deal with UTF-8 Byte Order Marks. The options are:&lt;/p&gt;
&lt;ul &gt;&lt;li&gt;&lt;i&gt;Add If Encoding Is UTF-8:&lt;/i&gt; always add a BOM when saving a file in UTF-8 encoding. Note that this will not work if the encoding is &lt;i&gt;System&lt;/i&gt;, as Qt Creator does not know what it actually is.&lt;/li&gt;
&lt;li&gt;&lt;i&gt;Keep If Already Present: &lt;/i&gt;save the file with a BOM if it already had one when it was loaded.&lt;/li&gt;
&lt;li&gt;&lt;i&gt;Always Delete:&lt;/i&gt; never write an UTF-8 BOM, possibly deleting a pre-existing one.&lt;/li&gt;&lt;/ul&gt;
&lt;p&gt;Note that UTF-8 BOMs are uncommon and treated incorrectly by some editors, so it usually makes little sense to add any.&lt;/p&gt;
&lt;p&gt;This setting does &lt;b&gt;not&lt;/b&gt; influence the use of UTF-16 and UTF-32 BOMs.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <item>
           <property name="text">
            <string>Add If Encoding Is UTF-8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Keep If Already Present</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Always Delete</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="utf8BomLabel">
          <property name="text">
           <string>UTF-8 BOM:</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBoxMouse">
       <property name="title">
        <string>Mouse and Keyboard</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QCheckBox" name="mouseHiding">
          <property name="text">
           <string>Hide mouse cursor while typing</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="mouseNavigation">
          <property name="text">
           <string>Enable &amp;mouse navigation</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="scrollWheelZooming">
          <property name="text">
           <string>Enable scroll &amp;wheel zooming</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="camelCaseNavigation">
          <property name="text">
           <string>Enable built-in camel case &amp;navigation</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="keyboardTooltips">
          <property name="toolTip">
           <string>Pressing Alt displays context-sensitive help or type information as tooltips.</string>
          </property>
          <property name="text">
           <string>Show help tooltips using keyboard shortcut (Alt)</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="helpTooltipsLabel">
          <property name="text">
           <string>Show help tooltips using the mouse:</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>30</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QComboBox" name="constrainTooltipsBox">
            <item>
             <property name="text">
              <string>On Mouseover</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>On Shift+Mouseover</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_2">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TextEditor::SimpleCodeStylePreferencesWidget</class>
   <extends>QWidget</extends>
   <header location="global">texteditor/simplecodestylepreferenceswidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>cleanWhitespace</tabstop>
  <tabstop>inEntireDocument</tabstop>
  <tabstop>cleanIndentation</tabstop>
  <tabstop>addFinalNewLine</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>cleanWhitespace</sender>
   <signal>toggled(bool)</signal>
   <receiver>inEntireDocument</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>87</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>353</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cleanWhitespace</sender>
   <signal>toggled(bool)</signal>
   <receiver>cleanIndentation</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>323</y>
    </hint>
    <hint type="destinationlabel">
     <x>134</x>
     <y>384</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
